docs/SmartStore.md

```diff
@@ -0,0 +1,233 @@
+
+
+Here is an example command to encode and load your remote storage volume secret key and access key in the kubernetes secret object: `kubectl create secret generic <secret_store_obj> --from-literal='s3_access_key=<access_key>' --from-literal='s3_secret_key=<secret_key>'`
+  
+
+## Creating a Standalone instance utilizing SmartStore 
+1. Confirm your S3-based storage volume path and URL.
+2. Confirm the name of the Splunk indexes being used with the SmartStore volume. 
+3. Pass the volume and index configuration into the environment through the Standalone Customer Resource specification.
+
+
+Example:
+
+```yaml
+apiVersion: enterprise.splunk.com/v1beta1
+kind: Standalone
+metadata:
+  name: <name>
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 1
+  smartstore:
+    volumes:
+    - name: <remote_volume_name>
+         path: <remote_volume_path>
+         endpoint: https://s3-<region>.amazonaws.com
+         secretRef: <secret_store_obj>
+    indexes:
+      - name: <index_name_1>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_2>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_3>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+```
+
+The SmartStore parameters will be placed into the required .conf files in an app. The app is named as `splunk-operator`. In the case of a standalone deployment, the app is located at `/opt/splunk/etc/apps/`
+ 
+ 
+ 
+## Creating an IndexCluster instance utilizing SmartStore
+1. Create an IndexCluster. See [Examples](Examples.md).
+2. Confirm the name of the Splunk indexes being used with the SmartStore volume. 
+   Note: SmartStore configuration is defined only on the Cluster master.
+3. Pass the volume and index configuration into the environment through the ClusterMaster Custom Resource spec.
+
+```yaml
+apiVersion: enterprise.splunk.com/v1beta1
+kind: ClusterMaster
+metadata:
+  name: <name>
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  smartstore:
+    volumes:
+    - name: <remote_volume_name>
+         path: <remote_volume_path>
+         endpoint: https://s3-<region>.amazonaws.com
+         secretRef: <secret_store_obj>
+    indexes:
+      - name: <index_name_1>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_2>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_3>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+```
+
+The SmartStore parameters will be placed into the required .conf files in an app. The app is named as `splunk-operator`. In case of a Indexer cluster deployment, the app is located on Cluster master at `/opt/splunk/etc/master-apps/`. 
```

Comment:
Should we touch upon the fact, somewhere, that an app with higher precedence can potentially overwrite the index & volume settings in the `splunk-operator` app, and hence needs to be taken care of by the Admin?

Reply:
Good point. I will add that.